### DISCUSSION TITLE
Feature Request - Client bundle blacklist

### DISCUSSION THREAD

**Discussion by TheRealFlyingCoder** [remix-run/remix#2902] [Proposals]
Just a thought, but similar to how we currently have a `*.server.ts` flag to keep local server modules out of the client build, it would be extremely handy to include a config option for npm modules that should never show up or run in the client bundle:

```
module.exports = {
  appDirectory: "app",
  browserBuildDirectory: "public/build",
  publicPath: "/build/",
  serverBuildDirectory: "server/build",
  devServerPort: 8002,
  clientBlacklist: [
    'firebase-admin',
    'prisma'
    'stripe'
  ]
};
```

Firebase is the most obvious use case here, while other server modules may just bloat the client bundle or fail gracefully behind the scenes, the firebase and firebase-admin modules don't bundle together at all.

The resulting developer experience for firebase is a dance of re-exporting every utility you need behind a `firebase.server.ts` file.

Beyond firebase, being able to explicitly define these bundles would offer a safety net for developers... And I have a feeling you could eventually push warnings in the remix dev console when a blacklisted server module is being used in the wrong plase

**Comment by AlonGvili** [comment]
maybe you can update the title to Feature Request - Client bundle denied list or something like that

**Comment by TheRealFlyingCoder** [comment]
> maybe you can update the title to Feature Request - Client bundle denied list or something like that

Well the terminology would be up to the team in the end :P Blacklist is good enough to get across what I mean though, it makes sense right?
